### PR TITLE
use gradle build for both navi and webservice setup

### PR DESCRIPTION
**Pull request by anupkumangodan**
Resolves #

## Description

Simplify the setup of both ui and webservice. The various dependencies of the components cause problems
depending upon what kind of platform/machine it is being installed in. Based on the state of the underlying
system like node version

## Proposed Changes

use gradle build to handle installation of all required dependencies including node.
Only pre-requisite is to have jdk installed. I have verified this using openjdk-11
-

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,10 +24,17 @@
...
-    "pretty-quick": "^2.0.1",
+    "pretty-quick": "^2.0.2",
     "validate-commit": "^3.4.0"
+  },
+  "dependencies": {
+    "babel": "^6.23.0",
+    "browserslist": "^4.14.5",
+    "caniuse-lite": "^1.0.30001142",
+    "ember": "^1.0.3",
+    "ember-cli-babel": "^7.22.1"
```
Comment (jkusa) [inline package.json:38]: Not sure we want these at the top level

File: packages/webservice/app/build.gradle.kts [hunk 2]
```diff
@@ -6,9 +6,19 @@ description = "app"
...
+    id("com.github.node-gradle.node") version "2.2.4"
+}
+
+node {
+    //if needed specify node version
+    //version = "8.9.4" 
+    //if needed specify npm version
+    //npmVersion = "6.13.4"
+    distBaseUrl = "https://nodejs.org/dist"
+    download = true
```
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:21]: This is block complete

File: packages/webservice/app/build.gradle.kts [hunk 3]
```diff
@@ -48,20 +58,18 @@ tasks.withType<KotlinCompile> {
 }
 
 tasks.register<NpmTask>("installUIDependencies") {
-    setArgs(listOf("ci"))
+    setArgs(listOf("install"))
```
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:61]: Can we revert this? `ci` mode allows use to install via the package-lock.json

File: packages/webservice/app/build.gradle.kts [hunk 4]
```diff
@@ -48,20 +58,18 @@ tasks.withType<KotlinCompile> {
...
 
-tasks.register<Exec>("buildUI") {
-    dependsOn("installUIDependencies")
-
-    workingDir("../../..")
-
-    commandLine = listOf("npx", "lerna", "run", "prodbuild", "--scope", "navi-app", "--stream")
+tasks.register<NpmTask>("buildUI") {
+  dependsOn("installUIDependencies")
+  setArgs(listOf("run-script","demostart"))
```
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:69]: Does this start the navi-app via the node asset server?

File: packages/webservice/app/build.gradle.kts (hunk 3 above)
Comment (anupkumangodan) [reply packages/webservice/app/build.gradle.kts:61]: this was one of the problems i encountered while testing. The package-lock.json went out of sync and prevented the packages being installed/updated. we need something to  keep it updated/sync. Is there something already in place for it ?

File: packages/webservice/app/build.gradle.kts (hunk 4 above)
Comment (anupkumangodan) [reply packages/webservice/app/build.gradle.kts:69]: yes, i added demostart as a new script command to the main navi package.json. 

The original `lerna prodbuild` was causing memory errors in multiple machines even when I allocated 16 GB VM. Not sure why.

File: packages/webservice/app/build.gradle.kts (hunk 3 above)
Comment (anupkumangodan) [reply packages/webservice/app/build.gradle.kts]: reverted

File: packages/webservice/app/build.gradle.kts [hunk 5]
```diff
@@ -18,7 +27,7 @@ repositories {
 dependencies {
     implementation(project(":models"))
     implementation("org.springframework.boot:spring-boot-starter-security")
-    implementation("com.yahoo.elide", "elide-spring-boot-starter", "5.0.0-pr18")
+    implementation("com.yahoo.elide", "elide-spring-boot-starter", "5.0.0-pr15")
```
Comment (aklish) [inline packages/webservice/app/build.gradle.kts:30]: Why did you downgrade?  Also, latest version is 23.

File: package.json [hunk 6]
```diff
@@ -22,10 +23,17 @@
...
```
Comment (jkusa) [inline package.json:37]:
```suggestion
```

File: packages/webservice/app/build.gradle.kts [hunk 7]
```diff
@@ -6,9 +6,18 @@ description = "app"
...
     kotlin("jvm")
     kotlin("plugin.spring") version "1.3.72"
+    id("com.github.node-gradle.node") version "2.2.4"
+}
+
+node {
+    //if needed specify node version
+    //version = "8.9.4" 
+    //if needed specify npm version
+    //npmVersion = "6.13.4"
```
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:18]:
```suggestion
    npmVersion = "12.16.0"
```

File: packages/webservice/app/build.gradle.kts (hunk 5 above)
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:30]:
```suggestion
    implementation("com.yahoo.elide", "elide-spring-boot-starter", "5.0.0-pr23")
```

File: package.json [hunk 8]
```diff
@@ -10,6 +10,7 @@
     "url": "https://github.com/yahoo/navi"
   },
   "scripts": {
+    "build_prod": "lerna run prodbuild --scope navi-app --stream",
```
Comment (jkusa) [inline package.json:13]:
```suggestion
    "build-ui": "lerna run prodbuild --scope navi-app --stream",
```

File: packages/webservice/app/build.gradle.kts [hunk 9]
```diff
@@ -54,12 +63,9 @@ tasks.register<NpmTask>("installUIDependencies") {
...
 
-tasks.register<Exec>("buildUI") {
-    dependsOn("installUIDependencies")
-
-    workingDir("../../..")
-
-    commandLine = listOf("npx", "lerna", "run", "prodbuild", "--scope", "navi-app", "--stream")
+tasks.register<NpmTask>("buildUI") {
+  dependsOn("installUIDependencies")
+  setArgs(listOf("run-script","build_prod"))
```
Comment (jkusa) [inline packages/webservice/app/build.gradle.kts:68]:
```suggestion
  setArgs(listOf("run-script", "build-ui"))
```

File: packages/webservice/app/build.gradle.kts (hunk 4 above)
Comment (anupkumangodan) [reply packages/webservice/app/build.gradle.kts]: did not happen again. will log a bug if the memory error shows up. resolving for now.

File: packages/webservice/app/build.gradle.kts (hunk 5 above)
Comment (anupkumangodan) [reply packages/webservice/app/build.gradle.kts]: upgraded as per suggestion

File: package.json (hunk 1 above)
Comment (anupkumangodan) [reply package.json]: accepted and removed the pins

**Comment by anupkumangodan** [comment]
too much conflicts while rebasing to request-v2

**Pull request closed without merging**